examples/fishnet_api/main.py

```diff
@@ -446,13 +432,14 @@ async def set_dataset_available(dataset_id: str, available: bool) -> List[Datase
     ts_list = await Timeseries.fetch(dataset.timeseriesIDs)
     if not ts_list:
         raise HTTPException(status_code=424, detail="No Timeseries found")
-    ts_record = ts_list[0]
+
     # check if the timeseries is updated or not and only update the timeseries when it is needed
-    if ts_record.available != available:
-        ts_record.available = available
-        requests.append(ts_record.upsert())
-    else:
-        raise HTTPException(status_code=424, detail="Already updated")
+    for rec in ts_list:
+        if rec.available != available:
+            rec.available = available
+            requests.append(rec.upsert())
+        else:
+            raise HTTPException(status_code=200, detail="Already updated")
```

Comment:
No need to raise an exception at all, we let the code run to an end and execute all the requests that we racked up :+1: